### ISSUE TITLE
Chrome 49 Crash after couple update on the same point

### ISSUE THREAD

**Issue by sebastianbochan** [highcharts/highcharts#5193]
Problem appears only in the newest Chrome (49.0.2623).

Example:
- https://jsfiddle.net/w5ktfn9f/

We need to wait some time (like 3/4 minutes).

**Comment by TorsteinHonsi** [comment]
Here's a normalized demo without any random values: https://jsfiddle.net/highcharts/w5ktfn9f/1/

Still there is no regularity, sometimes it crashes after 20 iterations, sometimes 1000.

**Comment by TorsteinHonsi** [comment]
One workaround would be to set a slower update interval, as 10 ms is too fast to render anyway.

**Comment by RolandBanguiran** [comment]
You're right. There is no consistency in reproducing the crash issue.

Also, this seems to surface only on the recent Google Chrome update.

I tried to disable `scrollbar` and `navigator` and seems to resolve the issue.
But we needed this feature as our users are already used to it and liked it.

Reducing the timer speed for like `500ms` seems to reduce the crashing but still eventually will crash.

By the way, it seems that the whole chart `redraw` twice when the `navigator` is enabled. Please see my JSFiddle below.
https://jsfiddle.net/RolandBanguiran/017u7xaf/

If we can find a way to resolve this redundancy, I think will be able to resolve the issue without disabling the `navigator`.

**Comment by RolandBanguiran** [comment]
I made some changes to the JSFiddle above to try to make the chart `redraw` once only per series update.
https://jsfiddle.net/RolandBanguiran/0g2a6bzf/

Basically, I disable navigator `adaptToUpdatedData` and just manually update the navigator series using `setData`. Then call chart `redraw` after all series values are set.

It seems to reduce the chance of browser crash.

Please let me know if this is the right solution. Or will this create some unwanted side effects?

**Comment by TorsteinHonsi** [comment]
> Basically, I disable navigator adaptToUpdatedData and just manually update the navigator series using setData. Then call chart redraw after all series values are set.

That sounds like a good idea.

I spent a lot of time yesterday on resolving the redundant redraw because of this issue and #2266. In the `issue-2266` branch, the redundant redraw is gone, but we [still see crashes in Chrome 49](http://jsfiddle.net/highcharts/w5ktfn9f/6/). It may be that it is less frequent though.

**Comment by RolandBanguiran** [comment]
Thank you Torstein.

Seems a huge improvement in terms of performance esp. for very dynamic and interactive charts with multiple axis and lots of rendered objects like ours.

By the way, is it safe to use `issue-2266` branch? Or do we need to wait for the next release? When is it?

**Comment by TorsteinHonsi** [comment]
> Seems a huge improvement in terms of performance esp. for very dynamic and interactive charts with multiple axis and lots of rendered objects like ours.

Great! Yes, we're saving a lot of redundancy on redrawing.

> By the way, is it safe to use issue-2266 branch? Or do we need to wait for the next release? When is it?

Nope, not safe yet. We'll hopefully continue to test the branch next week, then we'll merge it into the master and reach release within a few weeks.

**Comment by RolandBanguiran** [comment]
Thanks Torstein. Will wait for this version release.

**Comment by TorsteinHonsi** [comment]
I spent a couple of hours on the Chrome crashes, and did some observations:

* OHLC series doesn't crash, even though it shows the same data and otherwise behaves exactly like candlestick.
* By simplifying the [path in Candlestick](https://github.com/highcharts/highcharts/blob/v4.2.4/js/parts/CandlestickSeries.js#L108-L127), for instance to show only a straight line from top to bottom, the crashes go away. The downside of course is that it is no longer a candlestick...

In summary, it seems like the crashes somehow relate to the overall complexity of the SVG. So disabling the navigator helps, but simplifying the candlesticks also help.

**Comment by TorsteinHonsi** [comment]
Chrome issue reported at https://bugs.chromium.org/p/chromium/issues/detail?id=606311 .

**Comment by RolandBanguiran** [comment]
I modified my version with this update and seems no longer crashing. You did it! Thank you.

So will this and the redundant redrawing fix be included in the next version? When can we expect its release?

**Comment by TorsteinHonsi** [comment]
Great! Both these fixes will be included in the next maintenance release. It will be out within a few weeks.
